wee_slack.py

```diff
@@ -1898,13 +1898,23 @@ class SlackUser(object):
     """
 
     def __init__(self, **kwargs):
-        # We require these two things for a vaid object,
+        # We require these two things for a valid object,
         # the rest we can just learn from slack
         self.identifier = kwargs["id"]
-        self.slack_name = kwargs["name"]
-        self.name = kwargs["name"]
+        self.profile = {}  # in case it's not in kwargs
         for key, value in kwargs.items():
             setattr(self, key, value)
+
+        if self.profile.get("display_name"):
+            self.slack_name = self.profile["display_name"]
+            # TODO: configuration options here for 'what character to replace with'
+            # and 'whether to use normalized display name or real one'.
```

Comment:
Is there any reason for including an option to use the normalized display name? I mean, it's often good to include configuration options, but too many options will just be impractical, both for the user and the code.

Reply:
I have a few regulars on one of my slacks who have names that start with decorative non-latin characters, which makes tab-completing them difficult; I at least would use such an option. I'll take out the TODO, though -- if other people want it someone will file an issue.

Reply:
Okay, didn't think about tab completion. Fair enough.